app/src/main/java/cm/aptoide/pt/app/view/AppViewPresenter.java

```diff
@@ -199,21 +199,21 @@ private void sendAppViewLoadAnalytics(AppViewModel appViewModel) {
   }
 
   public Observable<AppViewModel> loadAds(AppViewModel appViewModel) {
-    return Observable.mergeDelayError(loadInterstitialAds(), loadOrganicAds(appViewModel),
-        loadBannerAds())
+    return Observable.mergeDelayError(loadInterstitialAds(appViewModel.getAppModel()),
+        loadOrganicAds(appViewModel), loadBannerAds(appViewModel.getAppModel()))
         .map(__ -> appViewModel)
         .onErrorReturn(throwable -> {
           crashReport.log(throwable);
           return appViewModel;
         });
   }
 
-  private Observable<Boolean> loadInterstitialAds() {
+  private Observable<Boolean> loadInterstitialAds(AppModel appModel) {
     return appViewManager.shouldLoadInterstitialAd()
         .observeOn(viewScheduler)
         .flatMap(shouldLoad -> {
           if (shouldLoad) {
-            view.initInterstitialAd();
+            view.initInterstitialAd(appModel.isMature());
```

Comment:
this method only depends on the mature flag, we can remove the heavier dependency to not couple this method to the AppModel